Python/pystrtod.c

```diff
@@ -391,6 +391,8 @@ _Py_string_to_number_with_underscores(
     char *dup, *end;
     PyObject *result;
 
+    assert(s[orig_len] == '\0');
```

Comment:
I'm not sure about this assertion. For performance, it would make sense to call _Py_string_to_number_with_underscores() on a sub-string which doesn't end with a NUL character. If you want to add an assertion, please do it at the caller site.

Reply:
This assertion is fine to me. `_Py_string_to_number_with_underscores()` is always called with a substring that end with a NUL character (if it doesn't end, we make a copy and call this function with it). And `_Py_string_to_number_with_underscores()` calls `innerfunc()` which expects a NUL terminated string.

Reply:
Oh, the following lines calls strchr() which expects that the string ends with a NUL byte.

In that case, should we also test that strlen(s) == orig_len to block embedded NUL byte?

Reply:
It is not needed.  NUL byte in string is valid.

```
>>> len("\0\0\0")
3
```